### ISSUE TITLE
SchemaFor resolveDefault wrongfully returns toString

### ISSUE THREAD

**Issue by oriy** [sksamuel/avro4s#229]
```
  implicit object MyClassToSchema extends ToSchema[MyClass] {
    override val schema: Schema = ToSchema.LongToSchema()
  }

  implicit object MyClassToValue extends ToValue[MyClass] {
    override def apply(value: MyClass): Long = value.getUnderlyingValue()
  }
```

```
@AvroNamespace("com.me")
final case class TestDefault(field: MyClass = MyClass.defaultValue)
```

```
{
  "type" : "record",
  "name" : "TestDefault",
  "namespace" : "com.me",
  "fields" : [ {
    "name" : "field",
    "type" : "long",
    "default" : "0"
  } ]
}
```

given the overridden schema, we got an invalid default value type 

 - schema = (Schema$LongSchema) "long"
 - defaultValue = **(TextNode) "0"**

expected `ToValue` to be applied on the default value
 - schema = (Schema$LongSchema) "long"
 - defaultValue = **(LongNode) 0**

**Comment by oriy** [comment]
@sksamuel used `1.8.0`
looking at `master`, issue still exists (solved for float, exists for the general record)

**Comment by sksamuel** [comment]
In order to make this work, the schema generation macros would have to rely on an instance of `Encoder` (or `ToValue` as it was called < 2.0) in order to convert from the Scala default value to the appropriate Avro value. I'm not sure this is worth it, the schema macros are nicely self contained at the moment, not caring about encoding/decoding values.

You could workaround this by having a custom `SchemaFor` for the containing class, where you can set default value to anything you like. What do you think ?

**Comment by oriy** [comment]
I understand your concerns relying on the Encoder
However, using custom `SchemaFor` would be very painful

My first instinct was looking for an annotation such as `AvroType` or `AvroDefaultValue` with parameters of either default value or default value class suggesting a converter

**Comment by sksamuel** [comment]
I'd be open to an annotation @AvroDefaultValue which you place on the field.

**Comment by sksamuel** [comment]
This is fixed now in master and will be in 2.0.2
